### PR TITLE
Feature global opt

### PR DESCRIPTION
**Pull request by DantongWang**
I think I almost finished the implementation of pswarm method.

### CODE REVIEW

**Comment by yannikschaelte** [comment]
looks good. please add a test, add pyswarm to .travis_pip_reqs.txt, and wrap the pyswarm import in a try-except.

**Comment by yannikschaelte** [comment]
just merged the master into the branch, let's see if the errors remain.

File: pypesto/optimize/optimizer.py [hunk 1]
```diff
@@ -425,3 +430,33 @@ def is_least_squares(self):
...
+            options = {'maxiter': 200}
+        self.options = options
+
+    @fix_decorator
+    @time_decorator
+    @objective_decorator
+    def minimize(self, problem, x0, index):
+        lb = problem.lb
+        ub = problem.ub
+        xopt, fopt = pso(problem.objective.get_fval,
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py]: you currently do not check ``method=='pso'`` at all.

**Comment by paulstapor** [comment]
:tada:
cool to have that one in! :)

**Pull request merged**